builtin/principals/local/Principals.go

```diff
@@ -54,7 +56,7 @@ func (p Principals) Get(ctx context.Context, payload []byte) (context.Context, [
 	}
 
 	if len(principals) == 0 {
-		return ctx, principals, fmt.Errorf("No more principals after trim for %s", local.User)
+		return ctx, principals, princsPkg.NewNotFoundError("local", "No more principals after trim for "+local.User)
```

Comment:
We don't need to add ldapPrinc.User to error, the logger already add it as a field.